src-tauri/src/database/commands.rs

```diff
@@ -0,0 +1,76 @@
+use std::{
+    fs::create_dir_all,
+    io::{Error, ErrorKind},
+    path::{Path, PathBuf},
+};
+
+use serde_json::Value;
+
+use crate::{error::user_error::UserValue, DB};
+
+use super::{db::DATA_ROOT_DIR, debug::SystemData};
+
+// Will, in future, return disk/remaining size
+// Just returns the directories that have been set up
+#[tauri::command]
+pub fn fetch_download_dir_stats() -> Vec<PathBuf> {
+    let lock = DB.borrow_data().unwrap();
+    lock.applications.install_dirs.clone()
+}
+
+#[tauri::command]
+pub fn delete_download_dir(index: usize) {
+    let mut lock = DB.borrow_data_mut().unwrap();
+    lock.applications.install_dirs.remove(index);
+    drop(lock);
+    DB.save().unwrap();
+}
+
+#[tauri::command]
+pub fn add_download_dir(new_dir: PathBuf) -> UserValue<(), Error> {
+    // Check the new directory is all good
+    let new_dir_path = Path::new(&new_dir);
+    if new_dir_path.exists() {
+        let dir_contents = new_dir_path.read_dir()?;
+        if dir_contents.count() != 0 {
+            return UserValue::Err(Error::new(
+                ErrorKind::DirectoryNotEmpty,
+                "Selected directory cannot contain any existing files",
+            ));
```

Comment:
This is incredibly bloated? We have too many wrappers, it should be possible to imply them or something.

Reply:
Nevermind then, I thought you  had created the Error object